src/v3.ts

```diff
@@ -0,0 +1,39 @@
+import prettier from 'prettier';
+import { OpenAPI3, PropertyMapper } from './types';
+
+export const WARNING_MESSAGE = `/**
+ * This file was auto-generated by swagger-to-ts.
+ * Do not make direct changes to the file.
+ */
+`;
+
+export const PRETTIER_OPTIONS: prettier.Options = { parser: 'typescript', singleQuote: true };
+
+export default function generateTypesV3(schema: OpenAPI3, propertyMapper?: PropertyMapper): string {
+  if (!schema.components || !schema.components.schemas) {
+    throw new Error(`🧐 Components Object missing from schema https://swagger.io/specification`);
+  }
+
+  // 1st pass: expand $refs first to reduce lookups & prevent circular refs
+  const expandedRefs = JSON.parse(JSON.stringify(schema.components.schemas), (_, node) => {
+    // TODO: if a node has a $ref, then convert "#/components/schemas/<$ref>" -> components['schemas']['<$ref>']
```

Comment:
First TODO: test if a node (a “part” of the object) has a `.$ref`. If it does, then change the string `#/components/schemas/some_thing` to `components['schemas']['some_thing']`.